cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,47 @@
+import { generateUser } from '../support/commands';
 /// <reference types='cypress' />
-
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
+  it('Enter valid data', () => {
+    // eslint-disable-next-line max-len
+    const { email, username, phone, lastname, address, gender, hobbies, subject } = generateUser();
+    cy.get('#firstName').type(username);
+    cy.get('#lastName').type(lastname);
+    cy.get('#userEmail').type(email);
+    cy.get('#userNumber').type(phone);
+    cy.get('#currentAddress').type(address);
+    cy.get('#genterWrapper > .col-md-9').type(gender);
+    cy.get('#hobbiesWrapper > .col-md-9').type(hobbies);
+    // eslint-disable-next-line cypress/unsafe-to-chain-command
+    cy.get('.subjects-auto-complete__value-container')
+      .type(subject);
+    cy.contains('#react-select-2-option-0', subject)
+      .click();
+    cy.get('#dateOfBirthInput')
+      .click();
+    cy.contains('select', 'October').select('November');
+    cy.contains('select', '2024').select('1990');
```

Comment:
This line is trying to select the year '1990' by finding a select element containing '2024'. This might fail if the current selected year is not '2024'. You should directly select the year without depending on the currently selected year.